### ISSUE TITLE
Failing auto fix for `derivable_impls`

### ISSUE THREAD

**Issue by remlse** [rust-lang/rust-clippy#10494]
### Summary

clippy fails to apply an auto fix correctly for the lint `derivable_impls`. Here's the code snippet:

```rust
enum Direction {
    North,
    East,
    South,
    West,
}
use std::collections::HashSet;

use Direction::*;

impl Default for Direction {
    fn default() -> Self {
        Direction::North
    }
}
```

clippy's message is:
```
this `impl` can be derived
for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derivable_impls
`#[warn(clippy::derivable_impls)]` on by defaultclippy[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2Fhome%2Fremo%2Frepos%2Faoc%2F16%2F01%2Fsrc%2Fsanta.rs)
santa.rs(11, 1): remove the manual implementation...
santa.rs(1, 1): ...and instead derive it...: `#[derive(Default)]
`
santa.rs(2, 5): ...and mark the default variant: `#[default]
`

### Reproducer

I tried this code:

```rust
<code>
```

I expected to see this happen:

Instead, this happened:

**Comment by remlse** [comment]
I'm having my first look at the clippy codebase (still reading the development chapter in the book) but his commit seems to have introduced auto suggestions for this lint for enums:

**Comment by remlse** [comment]
If I run `cargo clippy --fix --broken-code`, this is the result: (I've gotten rid of the imports, they don't seem to make a difference)
```rust
#[derive(Default)]
num Direction {
    #[default]
    North,
    East,
    South,
    West,
}

impl Default for Direction {
    fn default() -> Self {
        Direction::North
    }
}
```
the "e" is stripped from "enum"

**Comment by remlse** [comment]
If I add the `pub` keyword in front of `enum`, the "p" is stripped. However, If I add a newline at the start of the file, everything works fine. It seems to me that if the enum is on the first line of the file, the auto fix will remove the first character from the enum definition.

**Comment by J-ZhengLi** [comment]
hmmmm, I think this issue might belongs to `cargo` but I'm not sure.

Also, since this bug only happens when the enum/struct is on the first line makes it hard to test, because testing autofixing will need a `// run-rustfix` at the top of a test file...
